src/kudu/rpc/rpc-test.cc

```diff
@@ -1360,5 +1366,42 @@ TEST_P(TestRpc, TestCancellationMultiThreads) {
   client_messenger->Shutdown();
 }
 
+TEST_F(TestRpc, TestCallWithNormalTLS) {
+  FLAGS_enable_normal_tls = true;
+  FLAGS_authenticate_via_CN = true;
+  FLAGS_trusted_CNs = "myclient.com";
+
+  string client_certificate_file;
+  string client_private_key_file;
+  string server_certificate_file;
+  string server_private_key_file;
+  string rpc_ca_certificate_file;
+
+  ASSERT_OK(security::CreateTestSSLCertForClientAndServer(GetTestDataDirectory(),
+                                                     &client_certificate_file,
+                                                     &client_private_key_file,
+                                                     &server_certificate_file,
+                                                     &server_private_key_file,
+                                                     &rpc_ca_certificate_file));
+  // Set up server.
+  Sockaddr server_addr;
+  ASSERT_OK(StartTestServer(&server_addr, true, server_certificate_file,
+        server_private_key_file, rpc_ca_certificate_file));
+
+  // Set up client.
+  SCOPED_TRACE(strings::Substitute("Connecting to $0", server_addr.ToString()));
+  shared_ptr<Messenger> client_messenger;
+  ASSERT_OK(CreateMessenger("Client", &client_messenger, 1, true,
+      client_certificate_file, client_private_key_file, rpc_ca_certificate_file));
+
+  Proxy p(client_messenger, server_addr, server_addr.host(),
+          GenericCalculatorService::static_service_name());
+  ASSERT_STR_CONTAINS(p.ToString(), strings::Substitute("kudu.rpc.GenericCalculatorService@"
+                                                            "{remote=$0, user_credentials=",
+                                                        server_addr.ToString()));
```

Comment:
You also need to test or assert that `AuthenticateByCertificate` properly executed, and that the server and the client both have a consistent view of the other's identity.